recipes/gtsam/all/conanfile.py

```diff
@@ -99,30 +99,43 @@ def _configure_cmake(self):
             self._cmake.definitions["GTSAM_INSTALL_CPPUNITLITE"] = self.options.install_cppunitlite
             self._cmake.definitions["GTSAM_INSTALL_GEOGRAPHICLIB"] = self.options.install_geographiclib
             self._cmake.definitions["GTSAM_USE_SYSTEM_EIGEN"] = False
-            self._cmake.configure(source_folder=self._source_subfolder, build_folder=self._build_subfolder)
+            self._cmake.configure(build_folder=self._build_subfolder)
         return self._cmake
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
-            if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) < 15:
-                raise ConanInvalidConfiguration ("GTSAM requirews MSVC >= 15")
+        if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) < 15:
+            raise ConanInvalidConfiguration("GTSAM requirews MSVC >= 15")
 
     def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
         self.requires("boost/1.72.0")
         self.requires("eigen/3.3.7")
-        if(self.options.with_TBB):
+        if self.options.with_TBB:
             self.requires("tbb/2020.0")
-            self.options["tbb"].tbbmalloc = True
+            if not self.options["tbb"].tbbmalloc:
+                raise ConanInvalidConfiguration("GTSAM requires tbb.tbbmaloc = True")
```

Comment:
The drawback of doing this is that you have to manually specify tbb.tbbmaloc = True when you are compiling since it us false by default on the TBB recipe. In this case I think a hardcoded True is better because is the only possible option.

Reply:
It's a matter of opinion imho.
I'll leave it up to the reviewers of CCI.